### ISSUE TITLE
New cop to detect condition branches made redundant by an early exit

### ISSUE THREAD

**Issue by Zopolis4** [rubocop/rubocop#14042]
## Is your feature request related to a problem? Please describe.

I've encountered code in my codebase that, in the first branch of an if statement, will end up iterating to the next part of the loop, thus skipping all the code following it. However, there is still an else statement after the if, which I believe is redundant as the if will always result in an early exit, and thus there is no need for the remaining code to be wrapped in an else statement.

## Describe the solution you'd like

A cop that registers an offense on the following:
```ruby
# bad
if condition
  foo
  return # return, next, break, etc
else
  bar
end
```
and autocorrects it to this:
```ruby
# good
if condition
  foo
  return # return, next, break, etc
end

bar
```

## Describe alternatives you've considered

An addition/configuration to an already existing cop, such as `Style/Next` or `Style/RedundantCondition`, but I thought that would increase complexity from a rubocop development perspective and make things more confusing for users.

## Additional context

I looked through the existing cops and tried different configurations and I'm fairly certain this isn't already a feature, and I also checked existing issues for similar feature requests.

**Comment by Zopolis4** [comment]
not stale

**Comment by Zopolis4** [comment]
not stale, my pr fixes this and is ready to merge

**Comment by koic** [comment]
This seems like something that should be proposed in the style guide first.
https://github.com/rubocop/ruby-style-guide

**Comment by Zopolis4** [comment]
not stale, waiting on style guide decision
